processing/src/main/java/io/druid/query/BaseQuery.java

```diff
@@ -105,79 +103,40 @@ public static void checkInterrupted()
   }
 
   public static final String QUERYID = "queryId";
-  private final DataSource dataSource;
+
   private final boolean descending;
   private final Map<String, Object> context;
-  private final QuerySegmentSpec querySegmentSpec;
-  private volatile Duration duration;
 
   public BaseQuery(
-      DataSource dataSource,
```

Comment:
I think this change is inevitable because a query can now involve multiple data sources. If you're concerned with the compatibility with existing user-defined queries, I added ```SingleSourceBaseQuery``` for them which keeps the original APIs of BaseQuery.